src/serializer/LazyObjectsSerializer.js

```diff
@@ -105,6 +118,121 @@ export class LazyObjectsSerializer extends ResidualHeapSerializer {
     return getOrDefault(this._valueLazyIds, obj, () => this._lazyObjectIdSeed++);
   }
 
+  _isLazyObject(val: Value) {
+    // Currently, only support lazy raw object.
+    return this._isRawObject(val);
+  }
+
+  // Must keep in-sync with ResidualHeapSerializer logic.
+  _isObject(val: Value) {
+    return !(
```

Comment:
Confusing, at least the name `_isObject` is confusing. On one hand, it seems that a simple `val instanceof ObjectValue` might to the trick. On the other hand, an array and a function value are not objects?

Reply:
Yes, if you look ResidualHeapSerializer._serializeValue(), we do not call serializeValueObject() for array or functions and many other cases. So _isObject() here really means the value that is serialized via serializeValueObject() method in ResidualHeapSerializer. 
Any better approach/suggestion welcome.